unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/AvatarEditorHUD/Scripts/EmotesCustomization/EmotesCustomizationComponentController.cs

```diff
@@ -0,0 +1,442 @@
+using DCL;
+using DCL.Helpers;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace EmotesCustomization
+{
+    public interface IEmotesCustomizationComponentController : IDisposable
+    {
+        /// <summary>
+        /// Initializes the emotes customization controller.
+        /// </summary>
+        /// <param name="userProfile">User profile.</param>
+        /// <param name="catalog">Wearables catalog.</param>
+        void Initialize(UserProfile userProfile, BaseDictionary<string, WearableItem> catalog);
+    }
+
+    public class EmotesCustomizationComponentController : IEmotesCustomizationComponentController
+    {
+        internal const int NUMBER_OF_SLOTS = 10;
+        internal const string PLAYER_PREFS_EQUIPPED_EMOTES_KEY = "EquippedNFTEmotes";
+
+        internal DataStore_EmotesCustomization emotesCustomizationDataStore => DataStore.i.emotesCustomization;
+        internal BaseDictionary<(string bodyshapeId, string emoteId), AnimationClip> emoteAnimations => DataStore.i.emotes.animations;
+        internal BaseVariable<bool> isStarMenuOpen => DataStore.i.exploreV2.isOpen;
```

Comment:
inject the `DataStore` instance in the constructor to avoid an implicit dependency

Reply:
For the specific case of the use of the `DataStore` as static class, I find it justifiable because it is a type of data that we are currently using also in the tests implementation and it is not preventing us from a good use of them. In fact we are using this way in a lot of places. Anyway... maybe it would be a good point to discuss in next talks with the team 👍